cmd/dep/gopath_scanner.go

```diff
@@ -53,8 +53,9 @@ func (g *gopathScanner) InitializeRootManifestAndLock(rootM *dep.Manifest, rootL
 	}
 
 	g.origM = &dep.Manifest{
-		Constraints: g.pd.constraints,
-		Ovr:         make(gps.ProjectConstraints),
+		Constraints:  g.pd.constraints,
+		Ovr:          make(gps.ProjectConstraints),
+		PruneOptions: gps.PruneNestedVendorDirs,
```

Comment:
Sorry I'm late to the prune party and need a refresher course. When you say "pruning is opt-in" and also "we remove nested directories by default", I am not sure what will happen if someone doesn't explicitly set `PruneOptions`.

Basically I am concerned that if it's supposed to be a default, then it would be helpful to have it set by default in a constructor, rather than having to remember to set it everywhere as it's currently done in this PR. Some places aren't setting any `PruneOptions` (like the importers) and I can't really tell if they should, or if it doesn't matter because it's the default anyway.

If it's opt-in, do you mean that nothing happens (including removing nested vendor dirs) unless the user specifies a flag on the CLI or in the manifest? If so then why is it being set in code, that seems like a default, not opt-in...

None of this is criticism. I'm just confused. 😊